libutils/string_lib.c

```diff
@@ -1781,3 +1781,23 @@ bool StringMatchesOption(
     }
     return StringEqualN_IgnoreCase(supplied, longopt, length);
 }
+
+char *StringJoin(const Seq *const seq, const char *sep)
+{
+    assert(seq != NULL);
+
+    Writer *const writer = StringWriter();
+    const size_t len = SeqLength(seq);
+    for (size_t i = 0; i < len; i++)
```

Comment:
I'd do `i < (len -1)` writing the separator *after* every item and then writing the last item after the `for` loop without a separator. But YMMV.

Reply:
Yup, but that's easy to check upfront.